src/preamble.js

```diff
@@ -802,6 +802,78 @@ function createExportWrapper(name, fixedasm) {
 }
 #endif
 
+#if ABORT_ON_WASM_EXCEPTIONS
+// Instrument all the exported functions to:
+// - abort if an unhandled exception occurs
+// - throw an exception if someone tries to call them after the program has aborted
+// Set settings.ABORT_ON_WASM_EXCEPTIONS for more info.
+__ATINIT__.unshift(function () { 
```

Comment:
there is a function for this, `addOnInit`.

is this guaranteed to happen before all other constructors etc.? It may be safer to do it the way asyncify does, literally when the wasm exports are received, which is guaranteed to be before any code runs.

Reply:
I'm still worried about this. `Module["asm"]` is an internal detail that we may change from time to time. In particular it's possible that we'll optimize things so that we do not look at that object anymore (we've taken a reference to the export and saved it). Whereas the asyncify way operates on the wasm module exports as they just arrive, which is future-proof.

(Separately, good point that this runs before global ctors, I forgot that these run first.)

Another issue is that export name minification affects `Module["asm"]` too, doesn't it? For example when I build hello world with -O3 I get
```javascript
var _main = Module["_main"] = function() {
 return (_main = Module["_main"] = Module["asm"]["d"]).apply(null, arguments);
};
```
So it's minified as `"d"` in this build.

One option might be to disable export name minification with this option, but that is a little unfortunate.

Another option might be for `callMain` to do something special, like set `wrapperDepth` to negative infinity (so it will always be negative in checks, until a finally block in `callMain` undoes that)?

Reply:
> Another issue is that export name minification affects Module["asm"] too, doesn't it? 

Yeah, that's why I have `var mainFunc = Module["asm"]["main"]` which the minifier modifies to `var mainFunc = Module["asm"]["d"]` as well and it works - the same transform is applied everywhere :)

> Another option might be for callMain to do something special, like set wrapperDepth to negative infinity (so it will always be negative in checks, until a finally block in callMain undoes that)?

Oooh, nice one! Not coming for a JavaScript background I always forget about evil things like this 😎 - this is probably the most bulletproof way of handling this, but it does mean there will be an #ifdefed bit of code in callMain for this there. I'll whip it up and we can review.

Reply:
Changed this now. Let me know if you think there's anything else that needs doing.